Framework/Core/include/Framework/ASoA.h

```diff
@@ -140,22 +148,54 @@ class ColumnIterator : ChunkingPolicy
   /// Move the iterator to the next chunk.
   void nextChunk() const
   {
-    auto previousArray = std::static_pointer_cast<arrow_array_for_t<T>>(mColumn->chunk(mCurrentChunk));
-    mFirstIndex += previousArray->length();
+    auto chunkToUse = mColumn->chunk(mCurrentChunk);
+    if constexpr (std::is_same_v<arrow_array_for_t<T>,arrow::FixedSizeListArray>) {
+      chunkToUse = std::dynamic_pointer_cast<arrow::FixedSizeListArray>(chunkToUse)->values();
```

Comment:
Same comment as below. Actually the helpers could be the same, I guess.

Reply:
achieved minor simplification with introduction of ColumnIterator::getCurrentChunk()